modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesMessageMetadata.java

```diff
@@ -265,6 +267,19 @@ public void updateMetadata(String newDestination, String newExchangeName) {
         log.debug("updated andes message metadata id= " + messageID + " new destination = " + newDestination);
     }
 
+    /**
+     * Update metadata of message, after a change of the compression state of the message. This will change AMQP bytes
+     * representing metadata. IsCompressed will be set to the given value.
+     *
+     * @param newCompressedMessageValue new value to indicate if the message is compressed or not
+     */
+    public void updateMetadata(boolean newCompressedMessageValue) {
+        this.metadata = createNewMetadata(this.metadata, newCompressedMessageValue);
+        this.isCompressed = newCompressedMessageValue;
+        log.debug("updated andes message metadata id = " + messageID + ", compression state of the message is " +
```

Comment:
Lets put a if condition to check whether debug is enabled and then log the debug message. Its to avoid concatenation time if debug is disabled.